src/text/non_blocking/raw_text_reader.rs

```diff
@@ -25,16 +25,19 @@ use crate::IonType;
 
 const INITIAL_PARENTS_CAPACITY: usize = 16;
 
-// Represents reader state that can be carried over between calls to `next`.
-// Currently this is used to communicate state when `next` needs to return due to lack of data.
+// Represents the current actions being carried out by the reader.
 #[derive(PartialEq, Debug)]
 enum ReaderState {
-    // Any previous `next` calls completed successfully and we have no carried state.
+    // Ready to read any value, no existing errors, no partial parses.
     Ready,
-    // A previous `next` call was able to parse the struct field's name, but not the value.
-    StructFieldValue,
-    // A previous `next` call popped the container, but was unable to find an outer container's delimiter or end.
-    ContainerDelimiter,
+    // A previous `step_out` call failed to complete, but may have progressed through some values.
+    // We need to continue the step_out before doing anything else.
+    SteppingOut {
+        // The depth the reader is attempting to step-out to.
+        target_depth: usize,
+        // Whether the step_out process is looking for the parent container or not.
```

Comment:
Hm, I'm unsure what "looking for the parent container" means here. It sounds distinct from expecting the closing delimiter. Maybe you could add an example?